src/packages/utils/tests/request-coordinator.test.ts

```diff
@@ -0,0 +1,98 @@
+import assert from "assert";
+import { RequestCoordinator } from "../src/utils/request-coordinator";
+
+describe("request-coordinator", () => {
+  const noop = () => undefined;
+  let coordinator: RequestCoordinator;
+
+  beforeEach("instantiate RequestCoordinator", () => {
+    coordinator = new RequestCoordinator(0);
+  });
+
+  describe("stop()", () => {
+    it("should pause processing", () => {
+      coordinator.stop();
+
+      assert(coordinator.paused);
+    });
+
+    it("should not allow processing to be resumed", () => {
+      coordinator.stop();
+
+      assert.throws(
+        () => coordinator.resume(),
+        new Error("Cannot resume processing requests, Ganache is disconnected.")
+      );
+    });
+
+    it("should not allow new requests to be queued", async () => {
+      coordinator.stop();
+
+      await assert.rejects(
+        coordinator.queue(noop, this, []),
```

Comment:
Instead of `this` can you use something like `{}` or `null`? `this` is an actual thing and at a quick glance it seems like it matters.

Reply:
Absolutely - that makes a lot of sense.